app/file.go

```diff
@@ -830,13 +829,11 @@ func (t *UploadFileTask) preprocessImage() *model.AppError {
 	// For animated GIFs disable the preview; since we have to Decode gifs
 	// anyway, cache the decoded image for later.
 	if t.fileinfo.MimeType == "image/gif" {
-		gifConfig, err := gif.DecodeAll(io.MultiReader(bytes.NewReader(t.buf.Bytes()), t.teeInput))
-		if err == nil {
-			if len(gifConfig.Image) > 0 {
-				t.fileinfo.HasPreviewImage = false
-				t.decoded = gifConfig.Image[0]
-				t.imageType = "gif"
-			}
+		image, format, err := t.imgDecoder.Decode(io.MultiReader(bytes.NewReader(t.buf.Bytes()), t.teeInput))
```

Comment:
@agnivade should this be a call to `DecodeMemBounded`, or did I misunderstand your earlier comment?

Reply:
This is fine. I later realized that we cannot use `DecodeMemBounded` because we need to keep the image for later processing.

I got on a call with Allan and explained the background behind this.